system/jlib/jlzw.cpp

```diff
@@ -2456,19 +2456,23 @@ class CCompressedFile : implements ICompressedFileIO, public CInterface
                 throw MakeStringException(-1,"Partial row written at end of file %d of %d",ol,trailer.recordSize);
             }
             flush();
-            trailer.datacrc = trailer.crc;
-            if (setcrc) {
-                indexbuf.append(sizeof(trailer)-sizeof(trailer.crc),&trailer);
-                trailer.crc = crc32((const char *)indexbuf.toByteArray(),
-                                indexbuf.length(),trailer.crc);
-                indexbuf.append(trailer.crc);
-            }
-            else {
-                trailer.datacrc = 0;
-                trailer.crc = ~0U;
-                indexbuf.append(sizeof(trailer),&trailer);
+            //Avoid writing out a header/footer if the file is empty
+            if (indexbuf.length() != 0)
```

Comment:
would testing trailer.expandedSize == 0 be better and mean can avoid call to flush() (on line 2458)

Reply:
flush returns immediately, but I think you're right it would be clearer.